Makefile

```diff
@@ -290,6 +308,36 @@ clean:
 	@rm -f $(LIB_DIR)/mpxe-gen/src/*.pb-c.c
 	@rm -f $(MPXE_DIR)/protogen/*_pb2.py
 
+# Note: build.py relies on a lot of relative paths so it would be easier to just cd and execute command 
+.PHONY: codegen
+codegen: codegen_protos
+	@echo "Generating from templates..."
+	@cd $(CODEGEN_DIR) && python scripts/build.py 
+	@find $(CODEGEN_DIR)/out -type f \( -iname '*.[ch]' -o -iname '*.ts' \) | xargs -r clang-format -i -fallback-style=Google
+	@find $(CODEGEN_DIR)/out -type f \( -iname '*.go'  \) | xargs -r gofmt -w
+	@find $(CODEGEN_DIR)/out -name \*.h -exec cp {} libraries/codegen-tooling/inc/ \;
+
+# Note: build_dbc has same issue as build.by with local paths
+.PHONY: codegen_dbc
+codegen_dbc:
+	@echo "Generating DBC file"
+	@cd $(CODEGEN_DIR) && python scripts/build_dbc.py
+
+.PHONY: codegen_protos 
+codegen_protos:
+	@echo "Compiling protos..."
+	@mkdir -p $(CODEGEN_DIR)/genfiles
+	@protoc -I=$(CODEGEN_DIR)/schema --python_out=$(CODEGEN_DIR)/genfiles $(CODEGEN_DIR)/schema/can.proto
+
+.PHONY: codegen_test
+codegen_test: codegen
+	@echo "Testing codegen..."
+	@python -m unittest discover -s $(CODEGEN_DIR)/scripts
+
+.PHONY: mock_can_data
+mock_can_data: socketcan
+	@python3 $(CODEGEN_DIR)/mock_can_data.py
```

Comment:
This errors out for me because it can't find system_can.dbc - I think you either need to `cd $(CODEGEN_DIR)` here, or in the script use `codegen/system_can.dbc` as the dbc path.